vm/src/stdlib/itertools.rs

```diff
@@ -181,6 +184,141 @@ impl PyItertoolsTakewhile {
+                    args.args.len()
+                )))
+            }
+
+            2 => {
+                let (iter, stop): (PyObjectRef, PyObjectRef) = args.bind(vm)?;
+                let stop = if !stop.is(&vm.get_none()) {
+                    let stop = pyobject_to_usize(stop, &vm);
+                    if !stop.is_some() {
+                        return Err(vm.new_value_error("Stop argument for islice() must be None or an integer: 0 <= x <= sys.maxsize.".to_string()));
+                    }
+                    stop
+                } else {
+                    None
+                };
+
+                (iter, 0usize, stop, 1usize)
+            }
+            _ => {
+                let (iter, start, stop, step): (
+                    PyObjectRef,
+                    PyObjectRef,
+                    PyObjectRef,
+                    PyObjectRef,
+                ) = args.bind(vm)?;
+                let stop = if !stop.is(&vm.get_none()) {
+                    let stop = pyobject_to_usize(stop, &vm);
+                    if !stop.is_some() {
+                        return Err(vm.new_value_error("Stop argument for islice() must be None or an integer: 0 <= x <= sys.maxsize.".to_string()));
+                    }
+                    stop
+                } else {
+                    None
+                };
+
+                let start = if !start.is(&vm.get_none()) {
+                    let start = pyobject_to_usize(start, &vm);
+                    if !start.is_some() {
+                        return Err(vm.new_value_error("Indices for islice() must be None or an integer: 0 <= x <= sys.maxsize.".to_string()));
+                    }
+                    start.unwrap()
+                } else {
+                    0usize
+                };
+
+                let step = if !step.is(&vm.get_none()) {
+                    let step = pyobject_to_usize(step, &vm);
+                    if !step.is_some() {
+                        return Err(vm.new_value_error(
+                            "Step for islice() must be a positive integer or None.".to_string(),
+                        ));
+                    }
+                    step.unwrap()
+                } else {
+                    1usize
+                };
+
+                (iter, start, stop, step)
+            }
+        };
+
+        let iter = get_iter(vm, &iter)?;
+
+        Ok(PyItertoolsIslice {
+            iterable: iter,
+            cur: RefCell::new(0),
+            next: RefCell::new(start),
+            stop: stop,
+            step: step,
+        }
+        .into_ref(vm)
+        .into_object())
+    }
+
+    #[pymethod(name = "__next__")]
+    fn next(&self, vm: &VirtualMachine) -> PyResult {
+        while *self.cur.borrow() < *self.next.borrow() {
+            call_next(vm, &self.iterable)?;
+            *self.cur.borrow_mut() += 1;
+        }
+
+        if self.stop.is_some() && *self.cur.borrow() >= self.stop.unwrap() {
```

Comment:
You can avoid the `is_some`+`unwrap` here.
```rust
if let Some(stop) = self.stop {
  if *self.cur.borrow() >= stop {
    return Err;
  }
```
or maybe
```rust
if self.stop.map_or(false, |stop| *self.cur.borrow() >= stop) {
  return Err;
}
```
or
```rust
if self.stop.map(|stop| *self.cur.borrow() >= stop).unwrap_or(false) {
  return Err;
}
```

Reply:
Liked the first one